R/palettes.R

```diff
@@ -122,3 +122,48 @@ show_ncaa_col <- function(...) {
           axis.ticks.x = element_blank())
 }
 
+#' Displays palettes for all teams for a specified sport
+#' @import ggplot2
+#' @importFrom dplyr filter
+#' @param sport character vector (basketball, soccer, football, hockey)
+#' @param ... arguments passed to other functions
+#' @export
+#' @seealso \code{\link[scales]{show_col}}
+#' @examples 
+#' show_sport_col(sport = "soccer")
+show_sport_col <- function(sport, ...){
+ sport <- tolower(sport)
+  
+  if(sport == "basketball"){
+    select_league <- c("nba", "wnba")
+  } else if(sport == "soccer") {
+    select_league <- c("epl", "mls", "nwsl")
+  } else if(sport == "football"){
+    select_league <- c("nfl")
+  } else if(sport == "nhl"){
+    select_league <- c("nhl")
+  } else {
+    stop("Invalid Sport")
+  }
+  
+  x <- teamcolors::teamcolors %>%
+    dplyr::filter(league %in% select_league) 
+  if (length(list(...)) > 0) {
+    x <- teamcolors::teamcolors %>%
+      dplyr::filter(...)
+  }
+  
+  ggplot(x, aes(x = name, color = name, fill = name)) + 
```

Comment:
Hmm. I see what you are getting are here, but I'm not sure it's necessary. If the passing of the dots on line 153 actually worked, then we could just have one (or maybe two) functions and the user could just write their own `filter()` condition. But it doesn't work and I'm not sure why. But I think that fixing that is the optimal solution. 

One major problem here is that we have essentially the same `ggplot` code in three different functions. At a minimum, I'd recommend having a standalone function called `show_col()` that would show one set of colors for any given subset of `teamcolors`. Then, the functions `show_league_col()` and whatever else, could do the appropriate filtering, call `show_col()`, and then add a `facet_wrap()`. 

But maybe that is my problem and not yours, @adror1 . If you can fix up the comments above I'm happy to accept this PR. Thanks!